serveradmin/serverdb/query_committer.py

```diff
@@ -206,17 +206,17 @@ def _delete_servers(changed, deleted, deleted_servers):
 def _create_servers(attribute_lookup, created):
     created_servers = {}
     for attributes in created:
-        if 'hostname' not in attributes:
+        if 'hostname' not in attributes or not attributes['hostname']:
```

Comment:
This would also be a possible solution and might be easier to read:
```suggestion
        if not attributes.get('hostname'):
```